tests/test_loader.py

```diff
@@ -11,35 +11,46 @@
 from .common import MockModule, async_get_persistent_notifications, mock_integration
 
 
-async def test_component_dependencies(hass: HomeAssistant) -> None:
```

Comment:
Why is this test removed?

Reply:
It is not removed, it is renamed.

Reply:
To clarify: the test `test_component_dependencies` was not removed, it was in fact extended to cover additional cases, which were not working properly under previous implementation, but are behaving correctly under this PR.
To keep the test more readable it was split into two parts: `test_circular_component_dependencies`, `test_nonexistent_component_dependencies`.